snuba/split.py

```diff
@@ -9,86 +9,155 @@
+        ):
+            return col_split(*args, **kwargs)
         if (
             use_split and limit and not body.get('groupby')
             and orderby[:1] == ['-timestamp']
             and remaining_offset < 1000
         ):
-            overall_result = None
-            split_end = to_date
-            split_start = max(split_end - timedelta(seconds=split_step), from_date)
-            total_results = 0
-            status = 0
-            while split_start < split_end and total_results < limit:
-                body['from_date'] = split_start.isoformat()
-                body['to_date'] = split_end.isoformat()
-                # Because its paged, we have to ask for (limit+offset) results
-                # and set offset=0 so we can then trim them ourselves.
-                body['offset'] = 0
-                body['limit'] = limit - total_results + remaining_offset
-                result, status = query_func(*args, **kwargs)
-
-                # If something failed, discard all progress and just return that
-                if status != 200:
-                    overall_result = result
-                    break
-
-                if overall_result is None:
-                    overall_result = result
-                else:
-                    overall_result['data'].extend(result['data'])
-
-                if remaining_offset > 0 and len(overall_result['data']) > 0:
-                    to_trim = min(remaining_offset, len(overall_result['data']))
-                    overall_result['data'] = overall_result['data'][to_trim:]
-                    remaining_offset -= to_trim
-
-                total_results = len(overall_result['data'])
-
-                if total_results < limit:
-                    if len(result['data']) == 0:
-                        # If we got nothing from the last query, expand the range by a static factor
-                        split_step = split_step * STEP_GROWTH
-                    else:
-                        # If we got some results but not all of them, estimate how big the time
-                        # range should be for the next query based on how many results we got for
-                        # our last query and its time range, and how many we have left to fetch.
-                        remaining = limit - total_results
-                        split_step = split_step * math.ceil(remaining / float(len(result['data'])))
-
-                    # Set the start and end of the next query based on the new range.
-                    split_end = split_start
-                    try:
-                        split_start = max(split_end - timedelta(seconds=split_step), from_date)
-                    except OverflowError:
-                        split_start = from_date
-
-            return overall_result, status
+            return time_split(*args, **kwargs)
         else:
             return query_func(*args, **kwargs)
+
+    def time_split(*args, **kwargs):
```

Comment:
I didn't review this specific function — I assume it didn't change?

Reply:
Yeah, just moved. Wish github had a way of displaying this.